src/Interpreters/ITokenExtractor.cpp

```diff
@@ -294,25 +298,81 @@ void SplitTokenExtractor::substringToGinFilter(const char * data, size_t length,
             gin_filter.addTerm(data + token_start, token_len);
 }
 
+std::vector<String> LogTokenExtractor::getTokens(const char * data, size_t length) const
+{
+    return ITokenExtractor::getTokens(data, length);
+}
```

Comment:
I that `getTokens` for `SplitTokenExtractor` and `LogTokenExtractor` is implemented just to call `ITokenExtractor::getTokens`, which seems extra code without a good reason for me. Is there any good reason to do it?

Reply:
The tokenizers have two APIs
- the iterator-style one (`nextInString*`), its faster and preferred
- `getTokens` which is a slower fallback.

There will be another tokenizer ([this one](https://github.com/ClickHouse/ClickHouse/pull/80174), wip) which only supports the latter.

For all other tokenizers, the fallback is implemented in terms of the iterator-style call.

I think we'll need a "capabilities" interface (e.g. static bools) which the caller can use to call the faster of both interfaces but that will be in a future PR.